backend/service/topology/querybuilder_test.go

```diff
@@ -0,0 +1,243 @@
+			id: "Search by column",
+			input: &topologyv1.SearchRequest_Filter{
+				Search: &topologyv1.SearchRequest_Filter_Search{
+					Field: "column.id",
+					Text:  "cat",
+				},
+			},
+			expect: "SELECT * FROM topology_cache WHERE quote_literal(id) LIKE $1",
+		},
+		{
+			id: "Search by Metadata",
+			input: &topologyv1.SearchRequest_Filter{
+				Search: &topologyv1.SearchRequest_Filter_Search{
+					Field: "metadata.label",
+					Text:  "cat",
+				},
+			},
+			expect: "SELECT * FROM topology_cache WHERE quote_literal(metadata->>'label') LIKE $1",
+		},
+		{
+			id: "Search all options",
+			input: &topologyv1.SearchRequest_Filter{
+				Search: &topologyv1.SearchRequest_Filter_Search{
+					Field: "metadata.label",
+					Text:  "cat",
+				},
+				TypeUrl: "type.googleapis.com/clutch.aws.ec2.v1.AutoscalingGroup",
+				Metadata: map[string]string{
+					"label":  "value",
+					"label2": "value2",
+				},
+			},
+			expect: "SELECT * FROM topology_cache WHERE quote_literal(metadata->>'label') LIKE $1 AND resolver_type_url = $2 AND metadata @> $3::jsonb",
+		},
+	}
+
+	for _, test := range testCases {
+		selectBuilder := sq.StatementBuilder.PlaceholderFormat(sq.Dollar).
+			Select("*").
+			From("topology_cache")
+
+		output, err := filterQueryBuilder(selectBuilder, test.input)
+		assert.NoError(t, err)
+
+		sql, _, err := output.ToSql()
+		assert.NoError(t, err)
+		assert.Equal(t, test.expect, sql)
+	}
+}
+
+func TestSortQueryBuilder(t *testing.T) {
+	testCases := []struct {
+		id     string
+		input  *topologyv1.SearchRequest_Sort
+		expect string
+	}{
+		{
+			id:     "Default to ID ASC",
+			input:  &topologyv1.SearchRequest_Sort{},
+			expect: "SELECT * FROM topology_cache ORDER BY ID ASC",
+		},
+		{
+			id: "Sort by custom column and direction",
+			input: &topologyv1.SearchRequest_Sort{
+				Field:     "column.cat",
+				Direction: topologyv1.SearchRequest_Sort_DESCENDING,
+			},
+			expect: "SELECT * FROM topology_cache ORDER BY $1 DESC",
+		},
+		{
+			id: "Sort by custom metadata and direction",
+			input: &topologyv1.SearchRequest_Sort{
+				Field:     "metadata.meow",
+				Direction: topologyv1.SearchRequest_Sort_ASCENDING,
+			},
+			expect: "SELECT * FROM topology_cache ORDER BY metadata->>'meow' ASC",
+		},
+		{
+			id: "Sort by custom metadata deeply nested",
+			input: &topologyv1.SearchRequest_Sort{
+				Field:     "metadata.meow.iam.a.cat",
+				Direction: topologyv1.SearchRequest_Sort_ASCENDING,
+			},
+			expect: "SELECT * FROM topology_cache ORDER BY metadata->'meow'->'iam'->'a'->'cat' ASC",
+		},
+	}
+
+	for _, test := range testCases {
+		selectBuilder := sq.StatementBuilder.PlaceholderFormat(sq.Dollar).
+			Select("*").
+			From("topology_cache")
+
+		output := sortQueryBuilder(selectBuilder, test.input)
+		sql, _, err := output.ToSql()
+		assert.NoError(t, err)
+		assert.Equal(t, test.expect, sql)
+	}
+}
+
+func TestConvertMetadataToQuery(t *testing.T) {
+	testCases := []struct {
+		id     string
+		input  string
```

Comment:
add test for empty string input? i think it can happen if someone passes `column.` as field.